test/cpp/end2end/xds/xds_override_host_end2end_test.cc

```diff
@@ -516,21 +522,57 @@ TEST_P(OverrideHostTest, DifferentPerRoute) {
       default_route_config_.virtual_hosts(0).routes(0);
   SetListenerAndRouteConfiguration(
       balancer_.get(), BuildListenerWithStatefulSessionFilter(), route_config);
-  balancer_->ads_service()->SetEdsResource(
-      BuildEdsResource(EdsResourceArgs({{"locality0", {CreateEndpoint(0)}}})));
+  balancer_->ads_service()->SetEdsResource(BuildEdsResource(EdsResourceArgs(
+      {{"locality0", {CreateEndpoint(0), CreateEndpoint(1)}}})));
   WaitForAllBackends(DEBUG_LOCATION);
   // Disabled for "echo1" method
-  auto cookies = GetCookiesForBackend(
+  auto echo1_cookie = GetCookiesForBackend(
       DEBUG_LOCATION, 0, 1, RpcOptions().set_rpc_method(METHOD_ECHO1));
-  ASSERT_EQ(cookies.empty() ? "" : cookies.front().name, "");
+  ASSERT_EQ(echo1_cookie.empty() ? "" : echo1_cookie.front().name, "");
   // Overridden for "echo2" method
-  cookies = GetCookiesForBackend(DEBUG_LOCATION, 0, 1,
-                                 RpcOptions().set_rpc_method(METHOD_ECHO2));
-  ASSERT_EQ(cookies.empty() ? "" : cookies.front().name, kOverriddenCookie);
+  auto echo2_cookie = GetCookiesForBackend(
+      DEBUG_LOCATION, 0, 1, RpcOptions().set_rpc_method(METHOD_ECHO2));
+  ASSERT_EQ(echo2_cookie.empty() ? "" : echo2_cookie.front().name,
+            kOverriddenCookie);
   // Default for "echo" method
-  cookies = GetCookiesForBackend(DEBUG_LOCATION, 0, 1,
-                                 RpcOptions().set_rpc_method(METHOD_ECHO));
-  ASSERT_EQ(cookies.empty() ? "" : cookies.front().name, kCookieName);
+  auto echo_cookie = GetCookiesForBackend(
+      DEBUG_LOCATION, 0, 1, RpcOptions().set_rpc_method(METHOD_ECHO));
+  ASSERT_EQ(echo_cookie.empty() ? "" : echo_cookie.front().name, kCookieName);
+  // Echo1 endpoint ignores cookies
+  CheckRpcSendOk(DEBUG_LOCATION, 6,
+                 RpcOptions()
+                     .set_metadata({
+                         BuildCookieHeader(echo_cookie.front()),
+                         BuildCookieHeader(echo2_cookie.front()),
+                     })
+                     .set_rpc_method(METHOD_ECHO1));
+  EXPECT_EQ(backends_[0]->backend_service()->request_count(), 3);
```

Comment:
Please also check that the second backend got the other half of the requests.

Reply:
Done.